### PR TITLE
[trel] ignore anycast address as trel peer

### PR DESCRIPTION
**Pull request by mia1yang**

### CODE REVIEW

**Review by librasungirl — APPROVED** [review]

**Review by superwhd — APPROVED** [review]
LGTM 👍

File: src/trel_dnssd/trel_dnssd.cpp [hunk 1]
```diff
@@ -297,6 +297,12 @@ void TrelDnssd::OnTrelServiceInstanceAdded(const Mdns::Publisher::DiscoveredInst
     {
         otbrLogDebug("Peer address: %s", addr.ToString().c_str());
 
+        // Skip anycast (Refer to https://datatracker.ietf.org/doc/html/rfc2373#section-2.6.1)
+        if (addr.m64[1] == 0)
```
Comment (abtink) [inline src/trel_dnssd/trel_dnssd.cpp:301]: >```
>   |                         n bits                 |   128-n bits   |
>   +------------------------------------------------+----------------+
>   |                   subnet prefix                | 00000000000000 |
>   +------------------------------------------------+----------------+
>```

We are assuming the subnet prefix `n` will be `<= 64` (so the `m64[1]` will zero). Is `n>64` possible?

**Review by abtink — APPROVED** [review]
LGTM.
One question below:

File: src/trel_dnssd/trel_dnssd.cpp (hunk 1 above)
Comment (librasungirl) [reply src/trel_dnssd/trel_dnssd.cpp:301]: Generally on-link prefix allocated are of len 64 I think. we may have this assumption for now and take care of it if it happens somehow?

**Review by bukepo — APPROVED** [review]

File: src/trel_dnssd/trel_dnssd.cpp [hunk 2]
```diff
@@ -297,6 +297,12 @@ void TrelDnssd::OnTrelServiceInstanceAdded(const Mdns::Publisher::DiscoveredInst
...
+        {
+            continue;
+        }
+
         // If there are multiple addresses, we prefer the address
```
[the remark this answers is not in the document]
Comment (abtink) [reply src/trel_dnssd/trel_dnssd.cpp:306]: yes

**Review by jwhui — APPROVED** [review]

**Pull request merged**